exporters/memory/include/opentelemetry/exporters/memory/in_memory_span_exporter.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/exporters/memory/in_memory_span_data.h"
+
+using opentelemetry::exporter::memory::InMemorySpanData;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+/**
+ * A in memory exporter that switches a flag once a valid recordable was received
+ * and keeps track of all received spans in memory.
+ */
+class InMemorySpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+public:
+  InMemorySpanExporter();
+
+  std::unique_ptr<sdk::trace::Recordable> MakeRecordable() noexcept override;
+
+  sdk::trace::ExportResult Export(
+      const nostd::span<std::unique_ptr<sdk::trace::Recordable>> &recordables) noexcept override;
+
+  void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override
+  {};
+
+  std::vector<sdk::trace::SpanData> GetData() noexcept;
```

Comment:
I think it's better to return the complete `InMemorySpanData` from the getter. The exporter will be passed as an unique pointer to the processor and thus the user will not be able to call this getter.

Reply:
Updated